integration-cli/docker_cli_build_test.go

```diff
@@ -5660,3 +5660,434 @@ func (s *DockerSuite) TestBuildNullStringInAddCopyVolume(c *check.C) {
 	_, err = buildImageFromContext(name, ctx, true)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	args := []string{
+		"--build-arg", fmt.Sprintf("%s=%s", envKey, envVal),
+	}
+	dockerfile := fmt.Sprintf(`FROM busybox
+		ARG %s=
```

Comment:
I'm fine requiring `foo=""` instead of `foo=`, but I wouldn't want to allow `foo` for now specifically because we want to always have a default value, that is a requirement in the design, and although allowing an empty default value for `ARG foo` would make some sense, it could bite us in the future if, for some reason, we ever want to allow "no default values". It is just so much easier to keep `foo=` or if that's too *ugly*, then `foo=""`.